### PR TITLE
MULE-8216: Implement a validations extensions [For Mariano Gonzalez]

### PR DESCRIPTION
**Pull request by asequeira**
Adding more tests

### CODE REVIEW

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java [hunk 1]
```diff
@@ -149,22 +156,17 @@ public void isFalse() throws Exception
     public void notEmpty() throws Exception
     {
         final String flowName = "notEmpty";
-
```
Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: restore this empty line

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java [hunk 2]
```diff
@@ -79,21 +88,19 @@ public void url() throws Exception
...
-    private MuleEvent getTimeEvent(String time, String pattern) throws Exception {
+    private MuleEvent getTimeEvent(String time, String pattern) throws Exception
+    {
         MuleEvent event = getTestEvent(time);
         event.setFlowVariable("pattern", pattern);
-
         return event;
     }
 
+
```
Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: remove line

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java [hunk 3]
```diff
@@ -14,6 +14,7 @@
 import static org.junit.Assert.fail;
 import static org.mule.extension.validation.internal.ImmutableValidationResult.error;
 import static org.mule.extension.validation.internal.ValidationExtension.DEFAULT_LOCALE;
+
```
Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: remove empty line

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java [hunk 4]
```diff
@@ -193,8 +191,6 @@ public void successfulAll() throws Exception
     {
         MuleEvent event = getAllEvent(VALID_EMAIL, VALID_URL);
         MuleEvent responseEvent = runFlow("all", event);
-
-        // validation was successful and payload was untouched
```
Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: restore the comment

File: extensions/validation/src/test/java/org/mule/extension/validation/ValidationTestCase.java [hunk 5]
```diff
@@ -26,7 +26,6 @@
 
     static final String VALID_URL = "http://localhost:8080";
     static final String INVALID_URL = "here";
-
```
Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/ValidationTestCase.java]: restore line

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java (hunk 1 above)
Comment (asequeira) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: I don't think is a good practice to add empty lines in our code.

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java (hunk 4 above)
Comment (asequeira) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: We shouldn't add comments to the test, perhaps a better solution would be to change the method name to reflect what are we trying to prove.

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java (hunk 3 above)
Comment (asequeira) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: Is there a policy about which import statements allow an empty line?

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java (hunk 1 above)
Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: we use empty lines to separate sections on our tests and make it more readable all the time. Actually, the original PR passed code review in that form. If you check other test cases of other modules you'll find that all over the place

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java (hunk 3 above)
Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: intellij code stylesheet has the policy embedded. if it was intellij who inserted that line then fine

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java (hunk 4 above)
Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: agreed. change the name

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java (hunk 1 above)
Comment (asequeira) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: We should agree on a way for doing this and stick to it always or leave the decision to the committer.

Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: AFAIK you're the first to disagree. This is a common practice

File: extensions/validation/src/test/java/org/mule/extension/validation/ValidationTestCase.java (hunk 5 above)
Comment (asequeira) [inline extensions/validation/src/test/java/org/mule/extension/validation/ValidationTestCase.java]: I don't see why we should have lines between constants, a formatter tool will never understand that.

Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/ValidationTestCase.java]: the goal is to make it more readable to the human. No formatter tool covers 100% of the cases

File: extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java (hunk 1 above)
Comment (asequeira) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: I understand that for splitting setup, execution and assertions but not for declaring a variable. Anyway, we should agree to split these sections always or never do it.
For this particular case I'm creating a constant, which seems to be the correct solution. Is that OK for you?

Comment (marianogonzalez) [inline extensions/validation/src/test/java/org/mule/extension/validation/BasicValidationTestCase.java]: true that we should agree on something and always do it, although at least on the PR's I've review we always do it. If you want bring this up on post-scrum and we can hold the merge. 'Til then, I want my empty line :)

**Pull request closed without merging**